src/components/ExerciseHeader.tsx

```diff
@@ -57,21 +58,29 @@ const ExerciseHeader = ({
   confirmClose = true,
   labelOverride,
   isCloseButton = true,
+  exerciseKey,
 }: ExerciseHeaderProps): JSX.Element => {
   const [isModalVisible, setIsModalVisible] = useState(false)
   const [isFeedbackModalVisible, setIsFeedbackModalVisible] = useState(false)
   const theme = useTheme()
   const showProgress = numberOfWords !== undefined && numberOfWords > 0 && currentWord !== undefined
   const progressText = showProgress ? `${currentWord + 1} / ${numberOfWords}` : ''
-
+  const isWordList = exerciseKey === 0 || exerciseKey === undefined
   useEffect(() => {
-    const renderHeaderLeft = () => (
-      <NavigationHeaderLeft
-        title={labelOverride ?? getLabels().general.header.cancelExercise}
-        onPress={confirmClose ? () => setIsModalVisible(true) : () => navigation.dispatch(closeExerciseAction)}
-        isCloseButton={isCloseButton}
-      />
-    )
+    const renderHeaderLeft = () =>
+      isWordList && numberOfWords === (currentWord ?? 0) + 1 ? (
+        <NavigationHeaderLeft
+          title={labelOverride ?? getLabels().results.action.backToWordlist}
+          onPress={navigation.goBack}
```

Comment:
Why is it now `navigation.goBack` instead of the previous `confirmClose ? () => setIsModalVisible(true) : () => navigation.dispatch(closeExerciseAction)`?

Reply:
because user should go back to word list screen where all words are listed

Reply:
otherwise it navigates to all exercises in a module

Reply:
Sorry but I don't really get why this is necessary now, this was at least not part of the Issue. I still think this should be reverted if there is not a good reason for it.